src/openrct2/GameState.cpp

```diff
@@ -313,8 +314,8 @@ void GameState::UpdateLogic(LogicTimings* timings)
     auto day = _date.GetDay();
 #endif
 
-    DateUpdate();
-    _date = Date(static_cast<uint32_t>(gDateMonthsElapsed), gDateMonthTicks);
+    gDate.Update();
+    _date = gDate;
```

Comment:
Ideally this just becomes `_date.Update();` when the access of the global is eliminated by using `Date::GetDate()`, otherwise we need to maintain two states of the same data.